### PR TITLE
Drop support for Node 4, 6, 7, and 9

### PR DESCRIPTION
**Pull request by SparshithNR**
1. only LTS node is supported.
2. Node 8, 10, 12 is added

### CODE REVIEW

**Comment by rwjblue** [comment]
Looks like we have some failing tests under Node 10/12, mind taking a look?

**Comment by SparshithNR** [comment]
@rwjblue Tests are breaking because of a bug in windows implementation of checking if symlink exists or not. I created an issue in the nodeJS project.
https://github.com/nodejs/node/issues/30538#issuecomment-555373746

**Comment by SparshithNR** [comment]
Fixed the issue by adding a function that will do `fs.accessSync` and `fs.statSync` to decide symlink is not broken.

File: tests/index.js [hunk 1]
```diff
@@ -463,7 +461,7 @@ describe('broccoli-funnel', function() {
 
       builder = new broccoli.Builder(node);
       return builder.build()
-        .then(() => rimraf(node.outputPath))
```
Comment (stefanpenner) [inline tests/index.js]: @SparshithNR can you remove the rimraf dependency

**Comment by stefanpenner** [comment]
working on this now.

**Pull request merged**